internal/cmd/gocdk/demo.go

```diff
@@ -89,52 +67,52 @@ func registerDemoCmd(ctx context.Context, pctx *processContext, rootCmd *cobra.C
 	}
 	demoAddCmd.Flags().BoolVar(&force, "force", false, "re-add even the demo even if it has already been added, overwriting previous files")
 	demoCmd.AddCommand(demoAddCmd)
+
 	rootCmd.AddCommand(demoCmd)
 }
 
 func listDemos() error {
-	// Compute a sorted slice of available demos for usage.
-	var avail []string
-	for _, demo := range allDemos {
-		avail = append(avail, demo.name)
-	}
-	sort.Strings(avail)
-	fmt.Println(strings.Join(avail, "\n"))
+	fmt.Println(strings.Join(allDemos, "\n"))
 	return nil
 }
 
 func addDemo(ctx context.Context, pctx *processContext, demoToAdd string, force bool) error {
+	moduleRoot, err := pctx.ModuleRoot(ctx)
+	if err != nil {
+		return xerrors.Errorf("demo add: %w", err)
+	}
 	for _, demo := range allDemos {
-		if demo.name == demoToAdd {
-			return instantiateDemo(pctx, demo, force)
+		if demo == demoToAdd {
+			return instantiateDemo(pctx, moduleRoot, demo, force)
 		}
 	}
-	return xerrors.Errorf("%q is not a supported demo; try 'demo list' to see available demos")
+	return xerrors.Errorf("%q is not a supported demo; try 'gocdk demo list' to see available demos")
 }
 
 // instantiateDemo does all of the work required to add a demo of a
 // portable API to the user's project.
 // TODO(rvangent): It currently copies a single source code file. It should
 // additionally iterate over existing biomes, adding a config entry and possibly
 // Terraform files.
-func instantiateDemo(pctx *processContext, demo *demoInfo, force bool) error {
+func instantiateDemo(pctx *processContext, moduleRoot, demo string, force bool) error {
 	logger := log.New(pctx.stderr, "gocdk: ", log.Ldate|log.Ltime)
-	logger.Printf("Adding %q...", demo.name)
+	logger.Printf("Adding %q...", demo)
```

Comment:
should this be using the `pctx` logger?

Reply:
Yes, but that is not checked in yet. I will resolve conflicts like this before submitting depending on the submit order.